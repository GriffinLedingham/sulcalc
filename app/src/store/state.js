import smogon from "../../../dist/setdex/smogon";
import pokemonPerfect from "../../../dist/setdex/pokemonPerfect";
import usage from "../../../dist/setdex/usage";
import { maxGen } from "sulcalc";

export default {
  sets: {
    smogon,
    pokemonPerfect,
    usage,
    custom: Array(maxGen + 1)
      .fill(null)
      .map(() => ({}))
  },
  enabledSets: {
    smogon: true,
    pokemonPerfect: false,
    usage: true,
    custom: true
  },
  longRolls: false,
  fractions: false,
  gen: maxGen,
  attacker: null,
  defender: null,
  field: null,
  overrideReport: null
};
