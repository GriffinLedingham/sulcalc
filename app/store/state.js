import smogon from "../../dist/setdex/smogon";
import pokemonPerfect from "../../dist/setdex/pokemonPerfect";
import usage from "../../dist/setdex/usage";
import { maxGen } from "sulcalc";

export default {
  sets: {
    smogon,
    pokemonPerfect,
    usage,
    custom: Array(maxGen + 1)
      .fill(undefined)
      .map(() => ({}))
  },
  enabledSets: {
    smogon: true,
    pokemonPerfect: false,
    usage: true,
    custom: true
  },
  longRolls: false,
  fractions: false,
  gen: maxGen,
  attacker: undefined,
  defender: undefined,
  field: undefined,
  overrideReport: undefined
};
