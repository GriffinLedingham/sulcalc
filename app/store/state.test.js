import state from "./state";

test("equals expected state structure", () => {
  const { sets, ...restState } = state;
  expect(restState).toMatchSnapshot();
});
