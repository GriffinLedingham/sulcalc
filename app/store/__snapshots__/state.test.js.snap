// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`equals expected state structure 1`] = `
Object {
  "attacker": null,
  "defender": null,
  "enabledSets": Object {
    "custom": true,
    "pokemonPerfect": false,
    "smogon": true,
    "usage": true,
  },
  "field": null,
  "fractions": false,
  "gen": 7,
  "longRolls": false,
  "overrideReport": null,
}
`;
